shortfin/python/shortfin_apps/llm/components/batching/batching_trait.py

```diff
@@ -0,0 +1,40 @@
+# Copyright 2025 Advanced Micro Devices, Inc.
+#
+# Licensed under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+from abc import ABC, abstractmethod
+from ..config_struct import ModelParams
+
+
+# BatchingTrait is an interface that ensures that the batching
+# engine always has the methods needed by the facade class.
+# This is used second in inheritance, usually with another base
+# class. Since this is not a concrete class and Python implements
+# a neat MRO, we can safely use multiple inheritance without worrying
+# about complications like the diamond problem.
+class BatchingTrait(ABC):
```

Comment:
Name something that lines up with with an abstract class, e.g. `AbstractBatching`. Trait feels more like a parameter than a controlling structure.

Reply:
@rsuderman will fix this up in a follow up PR that's stacked on top of it.